src/transformers/cache_utils.py

```diff
@@ -1198,6 +1202,12 @@ def update(
         key_states = key_states.to(k_out.dtype)
         value_states = value_states.to(v_out.dtype)
 
+        if k_out.device.type == "meta":
+            k_out = torch.zeros(*k_out.size(), device=key_states.device, dtype=key_states.dtype)
+            v_out = torch.zeros(*v_out.size(), device=value_states.device, dtype=value_states.dtype)
+            self.key_cache[layer_idx] = k_out
+            self.value_cache[layer_idx] = v_out
```

Comment:
IMO the else branch should do: 
````
        k_out = self.key_cache[layer_idx]
        v_out = self.value_cache[layer_idx]
```
not sure memory wise if k_out is erased first or if you allocate more memory at this point.